tests/vm_backup_test.go

```diff
@@ -200,63 +200,98 @@ var _ = Describe("[smoke] VM Backup", func() {
-				g.Expect(err).ToNot(HaveOccurred())
-				_, err = f.KvClient.AppsV1().ControllerRevisions(f.Namespace.Name).Get(context.Background(), vm.Spec.Preference.RevisionName, metav1.GetOptions{})
-				g.Expect(err).ToNot(HaveOccurred())
-			}, 2*time.Minute, 2*time.Second).Should(Succeed())
+		Context("with instancetypes and preferences", func() {
+			nsDelFunc := func() {
+				err := f.KvClient.VirtualMachineInstancetype(f.Namespace.Name).
+					Delete(context.Background(), instancetypeName, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				err = f.KvClient.VirtualMachinePreference(f.Namespace.Name).
+					Delete(context.Background(), preferenceName, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}
 
-			By("Creating backup")
-			err = f.RunBackupScript(timeout, backupName, "", "a.test.label=included", f.Namespace.Name, snapshotLocation, f.BackupNamespace)
-			Expect(err).ToNot(HaveOccurred())
+			clusterDelFunc := func() {
+				err := f.KvClient.VirtualMachineClusterInstancetype().
+					Delete(context.Background(), instancetypeName, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				err = f.KvClient.VirtualMachineClusterPreference().
+					Delete(context.Background(), preferenceName, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}
 
-			By("Deleting VM, instancetype and preference")
-			err = f.KvClient.VirtualMachineInstancetype(f.Namespace.Name).
-				Delete(context.Background(), instancetypeName, metav1.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			err = f.KvClient.VirtualMachinePreference(f.Namespace.Name).
-				Delete(context.Background(), preferenceName, metav1.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			ok, err := framework.DeleteVirtualMachineAndWait(f.KvClient, f.Namespace.Name, vm.Name)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(ok).To(BeTrue())
+			clusterCleanup := func() {
+				err := f.KvClient.VirtualMachineClusterInstancetype().
+					Delete(context.Background(), instancetypeName, metav1.DeleteOptions{})
+				if err != nil {
+					Expect(errors.IsNotFound(err)).To(BeTrue())
+				}
+				err = f.KvClient.VirtualMachineClusterPreference().
+					Delete(context.Background(), preferenceName, metav1.DeleteOptions{})
+				if err != nil {
+					Expect(errors.IsNotFound(err)).To(BeTrue())
+				}
+			}
 
-			// Wait until ControllerRevision is deleted
-			Eventually(func(g Gomega) metav1.StatusReason {
-				_, err := f.KvClient.AppsV1().ControllerRevisions(f.Namespace.Name).Get(context.Background(), vm.Spec.Instancetype.RevisionName, metav1.GetOptions{})
-				if err != nil && errors.ReasonForError(err) != metav1.StatusReasonNotFound {
-					return errors.ReasonForError(err)
+			DescribeTable("[test_id:10270]with instancetype and preference", Label("PartnerComp"), func(itFunc func() error, pFunc func() error, vmFunc func() error, delFunc func(), cleanupFunc func()) {
```

Comment:
test id should be for the ns scoped entry, need to create another id for cluster one